From 3cf8d0b17d295f1a448d11cfa53309084075680f Mon Sep 17 00:00:00 2001
From: Igal Liberman <igall@marvell.com>
Date: Sun, 12 Mar 2017 14:09:06 +0200
Subject: [PATCH 0894/1345] crypto: inside-secure: store cell-index value

commit  e82fceaadd612b8cb68b2f5f1eec9227cc249954 from
https://github.com/MarvellEmbeddedProcessors/linux-marvell.git

This patch stores the cell-index which is read from
the device-tree.

Change-Id: I04be04b37336d4457301edb1818ffc711074b2ef
Signed-off-by: Igal Liberman <igall@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/37581
Tested-by: iSoC Platform CI <ykjenk@marvell.com>
Reviewed-by: Hanna Hawa <hannah@marvell.com>
Reviewed-by: Omri Itach <omrii@marvell.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/crypto/inside-secure/safexcel.c |    6 ++++++
 drivers/crypto/inside-secure/safexcel.h |    2 ++
 2 files changed, 8 insertions(+)

diff --git a/drivers/crypto/inside-secure/safexcel.c b/drivers/crypto/inside-secure/safexcel.c
index 57c93c2..c34f48a 100644
--- a/drivers/crypto/inside-secure/safexcel.c
+++ b/drivers/crypto/inside-secure/safexcel.c
@@ -1040,6 +1040,12 @@ static int safexcel_probe(struct platform_device *pdev)
 		return PTR_ERR(priv->base);
 	}
 
+	ret = of_property_read_u32(dev->of_node, "cell-index", &priv->id);
+	if (ret) {
+		dev_err(dev, "failed to read cell-index property\n");
+		return ret;
+	}
+
 	priv->clk = of_clk_get(dev->of_node, 0);
 	if (!IS_ERR(priv->clk)) {
 		ret = clk_prepare_enable(priv->clk);
diff --git a/drivers/crypto/inside-secure/safexcel.h b/drivers/crypto/inside-secure/safexcel.h
index 8841148..6343bbf 100644
--- a/drivers/crypto/inside-secure/safexcel.h
+++ b/drivers/crypto/inside-secure/safexcel.h
@@ -597,6 +597,8 @@ struct safexcel_crypto_priv {
 		struct crypto_queue queue;
 		bool need_dequeue;
 	} ring[EIP197_MAX_RINGS];
+
+	int id;
 };
 
 struct safexcel_context {
-- 
1.7.9.5

